api/src/main/java/org/motechproject/nms/api/web/BaseController.java

```diff
@@ -191,6 +192,20 @@ protected State getStateForFrontLineWorker(FrontLineWorker flw, Circle circle) {
         State state = frontLineWorkerService.getState(flw);
 
         if (state == null && circle != null) {
+            state = getStateFromCircle(circle);
+        }
+
+        return state;
+    }
+
+
+
+
+    protected State getStateFromCircle(Circle circle) {
+
+        State state = null;
+
+        if (circle != null) {
             List<State> states = circle.getStates();
 
             if (states.size() == 1) {
```

Comment:
It looked to me we are not checking the deployment status and whitelist
status for the first time caller in NMS.

Raised a bug : https://applab.atlassian.net/browse/NMS-205

On Mon, Jun 15, 2015 at 7:58 AM, abhishek srivastava <
abhisrivastava.17016@gmail.com> wrote:

> If there are more than 1 state , in that case,if kk is deployed in any of
> these we need to handle it as deployed service.
>
> I shall update the code after verification from SRS.
>
> Thanks,
> Abhishek
> On Jun 14, 2015 6:06 PM, "Frank Huster" <notifications@github.com> wrote:
>
>> In api/src/main/java/org/motechproject/nms/api/web/BaseController.java
>> <https://github.com/motech-implementations/mim/pull/385#discussion_r32387157>
>> :
>>
>> > @@ -191,6 +192,20 @@ protected State getStateForFrontLineWorker(FrontLineWorker flw, Circle circle) {
>> >          State state = frontLineWorkerService.getState(flw);
>> >
>> >          if (state == null && circle != null) {
>> > +            state = getStateFromCircle(circle);
>> > +        }
>> > +
>> > +        return state;
>> > +    }
>> > +
>> > +
>> > +
>> > +
>> > +    protected State getStateFromCircle(Circle circle) {
>> > +
>> > +        State state = null;
>> > +
>> > +        if (circle != null) {
>> >              List<State> states = circle.getStates();
>> >
>> >              if (states.size() == 1) {
>>
>> What happens if there is more than one state?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/motech-implementations/mim/pull/385/files#r32387157>.
>>
>